### DISCUSSION TITLE
Build standalone CLIs

### DISCUSSION THREAD

**Discussion by mnapoli** [php/frankenphp#948] [Ideas]
This is both a question and a suggestion: can FrankenPHP be used to create standalone CLI binaries?

I.e. PHP CLIs that can be distributed without users having to install PHP on their machine?

I looked at the docs, issues, discussions, and the source code, and it seems the feature to create standalone binaries is really meant to run FrankenPHP commands (start a server, or run script via the `./my-binary php-cli ...` command.

I'd want to run a CLI script directly. E.g. imagine if Composer was packaged as an executable file: you could run `composer install` and `composer would be the binary packaged by FrankenPHP.

Let me know if this is possible right now, I missed it.

**Comment by withinboredom** [comment]
Yes, it is possible. See https://github.com/bottledcode/durable-php/tree/v2/cli which is a PoC framework I've been working on off-and-on for the last couple of years.

It builds a cli, that includes php, and uses frankenphp to execute user scripts for specific endpoints.

BUT, if you just want to bundle php scripts then see https://github.com/crazywhalecc/static-php-cli

**Reply by mnapoli** [reply]
Ohh I didn't know static-php-cli simply has this feature bulit-in, thanks!
